x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceIntegTests.java

```diff
@@ -121,6 +125,45 @@ public void testGetRoleDescriptorsIntersectionForRemoteCluster() throws IOExcept
         );
     }
 
+    public void testCrossClusterAccessWithInvalidRoleDescriptors() {
```

Comment:
This seemed like a high-level enough place to get (sort-of) end-to-end coverage on the failure case. Ideally, we'd instead do this against a cluster, with transport headers we set as part of the test, but I spent some time looking into that in the past and did not come up with a fully-working solution. Let me know if you prefer more in-depth coverage of this change.